pyvista/plotting/plotting.py

```diff
@@ -3758,17 +3758,12 @@ def update_scalar_bar_range(self, clim, name=None):
             self.mapper.scalar_range = clim
             return
 
-        # Use the name to find the desired actor
-        def update_mapper(mapper_helper):
-            mapper_helper.scalar_range = clim
-            return
-
         try:
-            for mh in self._scalar_bar_mappers[name]:
-                update_mapper(mh)
+            # use the name to find the desired actor
+            for mh in self.scalar_bars._scalar_bar_mappers[name]:
+                mh.scalar_range = clim
         except KeyError:
-            raise KeyError('Name ({}) not valid/not found in this plotter.')
-        return
+            raise ValueError(f'Name ({name!r}) not valid/not found in this plotter.') from None
```

Comment:
I've changed the error from `KeyError` to `ValueError` for semantics (and the fact that the message of a `KeyError` should traditionally contain the offending key). I've also added `raise ... from None` to suppress the (implementation detail) context of the original exception.

Reply:
TIL: `raise ... from X`

Reply:
+1 for the error type change